database/sql/util.go

```diff
@@ -218,6 +219,9 @@ func (s *sqlDatabase) updatePool(pool Pool, param params.UpdatePoolParams) (para
 		pool.Image = param.Image
 	}
 
+	// no check necessary, we have sane defaults
+	pool.RunnerPrefix = param.RunnerPrefix
```

Comment:
I am concerned that if we ever add another code path that will use this function and we forget to add proper sane defaults there as well, this will come back to bite us in the behind. I suggest you attach a function to the ```params.UpdatePoolParams{}``` called something like ```GetRunnerPrefix()``` that checks if the prefix is set, and sets a ```DefaultRunnerPrefix``` (let's avoid magic strings).

This way we can be sure that a prefix makes it into the store, and we won't have to care about it being empty when we retrieve an entity and look for a prefix.

Reply:
Here is an example for the bootstrap timeout:

https://github.com/cloudbase/garm/blob/main/params/params.go#L164-L169

Something similar to this could be done for the runner prefix.

Reply:
done

Reply:
I am so silly. I said ```params.UpdatePoolParams{}``` should have that function, but in fact this is more useful in ```params.Pool{}``` as that is actually acted upon when creating the actual runner.

Sorry about this. I don't know what I was thinking...

Reply:
This will allow you to just use ```pool.GetRunnerPrefix()``` here:

https://github.com/cloudbase/garm/pull/57/files#diff-0928304a4f9afb0bf84875a5a19d4d42692d7ed54843569166a233b61cbb7603R404

Reply:
In fact, you could then use ```pool.GetRunnerPrefix()```, anywhere you call ```pool.RunnerPrefix``` ensuring you'll always get a sane value.